custom_components/localtuya/pytuya/__init__.py

```diff
@@ -286,8 +295,8 @@ def detect_available_dps(self):
 
         return detected_dps
 
-    def set_dpsUsed(self, dpsUsed):
-        self.dpsUsed = dpsUsed
+    def add_dps_to_request(self, dps_index):
```

Comment:
I like the idea of this better, but it would be convenient if `dps_index` could be either a single index or a list to avoid having for-loops all over the place. Should be something along the lines of:

```python
def add_dps_to_request(self, dps_index):
    if isinstance(dps_index, int):
        self.dps_to_request[str(dps_index)] = None
    else:
        self.dps_to_request.update({str(index): None for index in dps_index})
```